qml/Mercury.qml

```diff
@@ -0,0 +1,33 @@
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+
+Item {
+    property bool isBusy: true
+    property string entry: ""
+    property string bookmarkImage: ""
+
+    function setArticle(article_url) {
+        isBusy = true;
+        makeRequest(article_url);
+    }
+
+    function makeRequest(url) {
+        var doc = new XMLHttpRequest();
+        entry = "";
+        doc.onreadystatechange = function() {
+            if (doc.readyState == XMLHttpRequest.DONE && doc.status == 200) {
+                var entry_text = JSON.parse(doc.responseText).content;
+                entry_text = entry_text.replace(new RegExp("\%20[0-9].*x[a-z]?", "g"), "");
+                entry_text = entry_text.replace(new RegExp("<img(?!\/)", "g"), "<img width='" + (mainWindow.width - 2*Theme.horizontalPageMargin) + "'");
+                entry_text = entry_text.replace(bookmarkImage, "");
+                entry = entry_text;
+                isBusy = false;
+            }
+        }
+
+        doc.open("GET", "https://mercury.postlight.com/parser?url=" + url);
+        doc.setRequestHeader("Content-Type", "application/json");
+        doc.setRequestHeader("x-api-key", "WQuwYvDFOpW5rwWvBy4DPUP4h2r1AMkaRI5VJNN8");
+        doc.send();
+    }
+}
```

Comment:
I am not sure that using side service for readability it is a good idea.
May be it's better to give possibility to choose inbox reader mode or side reader mode?

Reply:
It's a more lightweight solution (no need to load the whole webkit engine), faster and uses less data, but I can see why someone would want to have a choice. That would require having a settings page, but that's ok since we could move "Logout" button there, show some details about account or something.

Best case scenario would be using Pocket's service, but that requires getting the app approved by them somehow afaik.

Reply:
I have sent request to pocket service about api access to its readermode. I am waiting the answer